docs/source/x_ext.rst

```diff
@@ -1287,19 +1299,19 @@ Handshake dependencies
 In order to avoid system level deadlock both the |processor| and the |coprocessor| shall obey the following rules:
 
 * The ``valid`` signal of a transaction shall not be dependent on the corresponding ``ready`` signal.
-* Transactions related to an earlier part of the instruction flow shall not depend on transactions with the same ``hartid`` and ``id`` related to a later part of the instruction flow. The instruction flow is defined from earlier to later as follows:
+* The only allowed dependencies between interfaces for transactions with the same ``hartid`` and ``id``are:
 
-  * compressed transaction
-  * issue transaction
-  * register transaction
-  * commit transaction
+  * ``issue_req`` may depend on ``compressed_resp``
+  * ``register`` may depend on ``compressed_resp``, ``issue_resp``, and ``commit``
+  * ``commit`` may depend on ``issue_resp``
+  * ``result`` may depend on ``issue_resp``, ``register``, and ``commit``
```

Comment:
and also on compressed_resp

Reply:
that wouldn't be an issue, but what would be the scenario? Also, both are coprocessor signals, so it is not an interop issue between CPU and coprocessor, is it?

Reply:
In the original text I did not differentiate when _req and _resp, but just talked about the channels in general. If you would do that then we have that 'issue may depend on compressed' and therefore it seems logical that 'result may depend on compressed as well' as result also is allowed to depend on issue (and register and commit). 

I don't have a specific use case in mind, but excluding it seems weird.

Reply:
I tried to focus on the kind of interaction that can lead to lock ups because of different assumptions in CPU and coprocoessor. So e.g. issue_resp depending on compressed_resp would not be one of those, as those are both coprocessor signals.

A coprocessor should not start executing before receiving issue_resp. Thus, it will use issue_req.instr as the basis for getting to result. compressed_resp signals are not actually used. 
I think what you are saying is, that because e.g. register depends on issue, it automatically depends on compressed?

Reply:
Yes exactly. Also I think we should talk about 'transactions' as in the original text and not about specific signals as now you would be excluding the valid/ready signals as well as dependencies fully within the coprocessor and dependencies fully within the processor.

Reply:
changed back to full interfaces and noting examples for why the dependence exists. Marking as resolved for now.